### ISSUE TITLE
png with transparency/alpha channel wouldn't shown

### ISSUE THREAD

**Issue by r4mp** [fschutt/printpdf#84]
printpdf version: 0.3.3

If I try to add a png with transparency, the image wouldn't shown. Other png files works fine.

```
let mut image_file = File::open("test.png").unwrap();
let image = Image::try_from(image::png::PngDecoder::new(&mut image_file).unwrap()).unwrap();
image.add_to_layer(current_layer.clone(), Some(Mm(x)), Some(Mm(y)), None, Some(1.0), Some(1.0), Some(300.0));
```

**Comment by fschutt** [comment]
Can you upload the PNG file to GitHub?

As far as I know, images with an alpha channel aren't supported right now because that would require support for "soft masks" in the PDF, which isn't implemented yet.

Also see https://github.com/fschutt/printpdf/issues/61

**Comment by r4mp** [comment]
![wouldnt_shown](https://user-images.githubusercontent.com/2659081/101404044-6266e780-38d6-11eb-9d40-9604fafbfdcc.png)

It really seems to be the problem/reason you described.

**Comment by fschutt** [comment]
I'm putting soft masks on my todo list, it's not that hard to implement (you need to split out the alpha channel and then reference it as a separate greyscale image). Can't say when I'll get back to working on printpdf though.

**Comment by anhtumai** [comment]
for newcomer going here, this is the workaround https://github.com/fschutt/printpdf/issues/119#issuecomment-1120434233 (work only with Rgba color space)

**Comment by BXHlixiaodong** [comment]
Hi, I find [SoftMask](https://docs.rs/printpdf/latest/printpdf/struct.SoftMask.html) is ready, but how to use it? it has no constructor function.

**Comment by fschutt** [comment]
@BXHlixiaodong This is a stub API, it doesn't work yet

**Comment by mikeparisstuff** [comment]
@fschutt, I'm also interested in soft mask functionality and would be willing to help out if you'd like.  If interested, could you point out some relevant parts of the code where you think it would make sense to plug it in and I'll see if I can help out?

**Comment by mikeparisstuff** [comment]
I went ahead and submitted a PR that adds support for rgba8 images. I think it may be helpful to more people so ideally we can merge it, but I am as of now unblocked via my fork. Here is the diff https://github.com/fschutt/printpdf/pull/158

**Comment by fschutt** [comment]
Yeah, this is definitely fixed in `0.8`, soft masks are implemented. Can be closed once 0.8 releases.

**Comment by fschutt** [comment]
Fixed in 0.8.

https://crates.io/crates/printpdf
